### ISSUE TITLE
clarification on the usage of custom metric

### ISSUE THREAD

**Issue by hadisotudeh** [microsoft/FLAML#299]
Hi,

As mentioned in the documentation, one can set up his/her own custom metric.

I just don't understand why X_test and y_test are inputs of this function. Since the metric is gonna be minimized, should not it be only based on X_train and y_train? 

Below, you can see the function I use currently:
```
def custom_metric(X_test, y_test, estimator, labels,
                  X_train, y_train, weight_test=None, weight_train=None,
                  config=None, groups_test=None, groups_train=None,
                  ):
    y_pred = estimator.predict(X_train)
    return float(sum(abs(y_train.to_numpy() - y_pred.to_numpy())) / sum(y_train.to_numpy())), {}
```

I think the documentation is missing these explanations with an example.

**Comment by qingyun-wu** [comment]
Hi @hadisotudeh,

Thank you for your questions and suggestions. And sorry for the confusion. Although the names of the arguments are X_test and y_test, when this custom_metric is called, you are passing validation data to them, i.e., custom_metric(X_test = X_val, y_test=y_val, …). We are not asking you to provide the actual test data from your task.

The custom function you wrote is returning the training loss as the primary metric to minimize. You can use it if you really want to, but validation loss is usually considered a better metric than training loss to evaluate a model.

Please find an example of the custom metric here: https://github.com/microsoft/FLAML/blob/72caa2172dc6e27b25caa4962dc968bb8b05b0ca/test/automl/test_multiclass.py#L100
(Note that in this example, test_loss means validation loss, considering the X_test and y_test we passed to it are from validation data).

We will improve the documentation accordingly. 

Thank you!
